.github/workflows/cleanup-branch-deployment.yml

```diff
@@ -0,0 +1,44 @@
+name: Clean up branch deployment
+on:
+  delete
+
+jobs:
+  delete_latest_deployment_for_branch:
+    name: Delete latest deployment for branch
+    if: github.event.ref_type == 'branch'
```

Comment:
Dont think this is needed as it can only be branch or tag

Reply:
Yep, but we don't want to trigger on tag deletion